### PR TITLE
Allow users to manage ImageRepository objects with the same access level as Components

### PR DESCRIPTION
**Pull request by mmorhun**
Since we are switching to the new way of managing image repositories in Konflux via a separate `ImageRepositopry` resource instead of `Component` annotations, we need to grant users permissions to work with the new resource. `ImageRepositopry` should be created together with the `Component` (by UI) in case user wants to use default image repository option.
The permissions are set in sync with the `Component` permissions of each role.

[STONEBLD-2473](https://issues.redhat.com/browse/STONEBLD-2473)

### CODE REVIEW

**Review by rajivnathan — APPROVED** [review]

**Review by chmeliik — APPROVED** [review]

**Comment by mfrancisc** [comment]
@mmorhun looks good! 

Could you please also create a paired PR ( same branch name ) in the toolchain-e2e repo, updating the expected resources:
- [appstudio admin](https://github.com/codeready-toolchain/toolchain-e2e/blob/716ec177235b9b043eddcb967567f2806e8463a2/testsupport/tiers/checks.go#L1599)
- [appstudio maintainer](https://github.com/codeready-toolchain/toolchain-e2e/blob/716ec177235b9b043eddcb967567f2806e8463a2/testsupport/tiers/checks.go#L1708)
- [appstudio contributor](https://github.com/codeready-toolchain/toolchain-e2e/blob/716ec177235b9b043eddcb967567f2806e8463a2/testsupport/tiers/checks.go#L1802)

**Review by sadlerap — APPROVED** [review]

**Comment by mmorhun** [comment]
@mfrancisc could you merge this PR when the tests PR gets merged?

**Pull request merged**
